### PR TITLE
implement support for 'use_pip' in PythonPackage easyblock

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 1]
```diff
@@ -114,20 +117,31 @@ def __init__(self, *args, **kwargs):
         if not 'modulename' in self.options:
             self.options['modulename'] = self.name.lower()
 
-        if self.cfg.get('zipped_egg', False) and not self.cfg.get('use_easy_install', False):
-            raise EasyBuildError("Installing zipped eggs requires use_easy_install = True")
-
+        self.use_setup_py = False
         if self.cfg.get('use_easy_install', False):
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/pythonpackage.py]: add a debug output about what is chosen?

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 2]
```diff
@@ -114,20 +117,31 @@ def __init__(self, *args, **kwargs):
...
+
+            # don't auto-install dependencies
+            self.cfg.update('installopts', '--no-deps')
+
+            if self.cfg.get('zipped_egg', False):
+                self.cfg.update('installopts', '--egg')
 
         else:
+            self.use_setup_py = True
             self.install_cmd = "python setup.py install"
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/pythonpackage.py]: why not makes this one a constant?  It is more consistent?

**Comment by wpoely86** [comment]
retest this please

**Comment by dominikborkowski** [comment]
While on the subject of being able to leverage use_pip in PythonPackage, could this include support for wheel packages in addition to eggs? 

https://pypi.python.org/pypi/wheel

Our specific use case would be ability to deploy TensorFlow:

https://www.tensorflow.org/versions/0.6.0/get_started/os_setup.html#pip_install

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/generic/pythonpackage.py]: done below

**Comment by boegel** [comment]
@wpoely86: please re-review? It would be great to get this in (required for https://github.com/hpcugent/easybuild-easyconfigs/pull/2461, which is required for IPython 4.x (cfr. https://github.com/hpcugent/easybuild-easyconfigs/pull/2457), which would be nice to support in HOD).

**Comment by wpoely86** [comment]
lgtm

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Pull request merged**